rust/azure_iot_operations_mqtt/src/session.rs

```diff
@@ -115,11 +135,11 @@ impl From<SessionErrorKind> for SessionError {
 #[error(transparent)]
 pub struct SessionConfigError(#[from] adapter::ConnectionSettingsAdapterError);
 
+// NOTE: Retain a struct/kind pattern for `SessionExitError` for future-proofing
```

Comment:
`.attempted` could become relevant again in the future and be re-added if, for instance, we needed to have a "wait for connect and then end session" API of some kind. This would also potentially add new error cases, thus why the `SessionExitErrorKind` is non-exhaustive. I don't think these are likely scenarios, but it seems safer to not take too strong a position on exit errors until we see how customers use it. Plus, there's a symmetry with all "kinds" being non-exhaustive.